### PR TITLE
[Merged by Bors] - feat: part-preserving equivalence of an equipartition with `Fin s.card`

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
Part of #9317.

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Order/Partition/Equipartition.lean [hunk 1]
```diff
@@ -66,8 +77,94 @@ theorem IsEquipartition.card_part_le_average_add_one (hP : P.IsEquipartition) (h
   exact Finset.EquitableOn.le_add_one hP ht
 #align finpartition.is_equipartition.card_part_le_average_add_one Finpartition.IsEquipartition.card_part_le_average_add_one
 
-/-! ### Discrete and indiscrete finpartition -/
+theorem IsEquipartition.filter_neg_average_add_one_eq_average (hP : P.IsEquipartition) :
```
Comment (jcommelin) [inline Mathlib/Order/Partition/Equipartition.lean:80]:
```suggestion
theorem IsEquipartition.filter_ne_average_add_one_eq_average (hP : P.IsEquipartition) :
```

File: Mathlib/Order/Partition/Equipartition.lean [hunk 2]
```diff
@@ -66,7 +77,94 @@ theorem IsEquipartition.card_part_le_average_add_one (hP : P.IsEquipartition) (h
...
+
+/-- An equipartition of a finset with `n` elements into `k` parts has
+`n - n % k` parts of size `n / k`. -/
+theorem IsEquipartition.card_small_parts_eq_mod (hP : P.IsEquipartition) :
+    (P.parts.filter fun p ↦ p.card = s.card / P.parts.card).card =
+    P.parts.card - s.card % P.parts.card := by
+  conv_rhs =>
+    arg 1
+    rw [← filter_card_add_filter_neg_card_eq_card (p := fun p ↦ p.card = s.card / P.parts.card + 1)]
+  rw [hP.card_large_parts_eq_mod, add_tsub_cancel_left, hP.filter_neg_average_add_one_eq_average]
```
Comment (Parcly-Taxel) [inline Mathlib/Order/Partition/Equipartition.lean:110]:
```suggestion
  rw [hP.card_large_parts_eq_mod, add_tsub_cancel_left, hP.filter_ne_average_add_one_eq_average]
```

File: Mathlib/Order/Partition/Equipartition.lean (hunk 1 above)
Comment (Parcly-Taxel) [reply Mathlib/Order/Partition/Equipartition.lean]: :+1:

**Review by jcommelin — COMMENTED** [review]
Thanks :tada: 

If CI passes, please remove the label `awaiting-CI` and merge this yourself, by adding a comment `bors r+`.

bors d+

**Comment by Parcly-Taxel** [comment]
bors merge

**Pull request closed without merging**